### PR TITLE
Add more templates for Rbac

### PR DESCRIPTION
**Pull request by astrozzc**
Email templates added for each of the event types.

### CODE REVIEW

File: engine/src/main/resources/templates/Rbac/customRoleDeletedEmailTitle.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+{action.timestamp.toUtcFormat()} - Custom role {action.events[0].payload.name} has deleted.
```
Comment (gwenneg) [inline engine/src/main/resources/templates/Rbac/customRoleDeletedEmailTitle.txt:1]:
```suggestion
{action.timestamp.toUtcFormat()} - Custom role {action.events[0].payload.name} has been deleted.
```

File: engine/src/main/resources/templates/Rbac/customGroupCreatedEmailBody.html [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            <!-- Content block -->
+            <tr>
+                <td class="rh-content__block">
+                    <p>
+                        Hi,
+                        <br>
+                        <br>
+                        A custom group has been created by {action.events[0].payload.username}.
+                        <br>
+                        Check the created group <a class="rh-url" href="{environment.url('/settings/rbac/groups/detail/' + action.events[0].payload.uuid)}"><b>{action.events[0].payload.name}</b></a>
```
Comment (gwenneg) [inline engine/src/main/resources/templates/Rbac/customGroupCreatedEmailBody.html:40]: Unfortunately, `'/settings/rbac/groups/detail/' + action.events[0].payload.uuid` will be parsed as a string and the output will look like this:
```
https://console.stage.redhat.com/settings/rbac/groups/detail/'%20+%20action.events%5B0%5D.payload.uuid
```
Here's how to use an environment-based dynamic URL and append something at its end:
```suggestion
                        Check the created group <a class="rh-url" href="{environment.url('/settings/rbac/groups/detail/')}{action.events[0].payload.uuid}"><b>{action.events[0].payload.name}</b></a>
```

File: engine/src/main/resources/templates/Rbac/customGroupUpdatedEmailBody.html [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                <td class="rh-content__block">
+                    <p>
+                        Hi,
+                        <br>
+                        <br>
+                        Custom group has been updated by {action.events[0].payload.username}.
+                        <br>
+                        {#if action.events[0].payload.role}
+                        <p>
+                            The role <a class="rh-url" href="{environment.url('/settings/rbac/groups/detail/' + action.events[0].payload.role.uuid)}"><b>{action.events[0].payload.name} has been
```
Comment (gwenneg) [inline engine/src/main/resources/templates/Rbac/customGroupUpdatedEmailBody.html:42]:
```suggestion
                            The role <a class="rh-url" href="{environment.url('/settings/rbac/groups/detail/')}{action.events[0].payload.role.uuid}"><b>{action.events[0].payload.name} has been
```

File: engine/src/main/resources/templates/Rbac/customGroupUpdatedEmailBody.html [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                        {/if}
+                        {#if action.events[0].payload.principal}
+                        <p>
+                            The user {action.events[0].payload.principal} has been
+                            {action.events[0].payload.operation} {#if action.events[0].payload.operation == "added"}to{#else}from{/if} 
+                            the group.
+                        </p>
+                        {/if}
+                        <br>
+                        Check the updated group <a class="rh-url" href="{environment.url('/settings/rbac/groups/detail/' + action.events[0].payload.uuid)}"><b>{action.events[0].payload.name}</b></a>
```
Comment (gwenneg) [inline engine/src/main/resources/templates/Rbac/customGroupUpdatedEmailBody.html:55]:
```suggestion
                        Check the updated group <a class="rh-url" href="{environment.url('/settings/rbac/groups/detail/')}{action.events[0].payload.uuid}"><b>{action.events[0].payload.name}</b></a>
```

File: engine/src/main/resources/templates/Rbac/customPlatformGroupUpdatedEmailBody.html [hunk 5]
```diff
@@ -35,19 +35,18 @@
...
-                        {#if action.events[0].payload.operation != "delete"} Check the group <a class="rh-url" href="https://console.redhat.com/settings/rbac/groups/detail/{action.events[0].payload.uuid}/"><b>{action.events[0].payload.name}</b></a>{/if}
-                    </p>
-                    {#if action.events[0].payload.operation == "update" and action.events[0].payload.platform_default == "true"}
-                    <p>
-                        The {#if action.events[0].payload.system == "true"}Red Hat{/if} role{#if action.events[0].payload.roles.size() >1}s are{#else} is{/if} 
-                        {action.events[0].payload.role_operation}ed {#if action.events[0].payload.role_operation == "added"}to{#else}from{/if} 
-                        {#if action.events[0].payload.system != "true"}custom{/if} platform default access group.
+                        {#if action.events[0].payload.role}
+                        <p>
+                            The role <a class="rh-url" href="{environment.url('/settings/rbac/groups/detail/' + action.events[0].payload.role.uuid)}"><b>{action.events[0].payload.role.name} has been
```
Comment (gwenneg) [inline engine/src/main/resources/templates/Rbac/customPlatformGroupUpdatedEmailBody.html:42]:
```suggestion
                            The role <a class="rh-url" href="{environment.url('/settings/rbac/groups/detail/')}{action.events[0].payload.role.uuid}"><b>{action.events[0].payload.role.name} has been
```

File: engine/src/main/resources/templates/Rbac/customPlatformGroupUpdatedEmailBody.html [hunk 6]
```diff
@@ -35,19 +35,18 @@
...
+                            {action.events[0].payload.operation} {#if action.events[0].payload.operation == "added"}to{#else}from{/if} 
+                            the group.
+                        </p>
+                        {/if}
                         <br>
-                        And the role list is {#for role in action.events[0].payload.roles}role{/for}.
+                        Check the updated group <a class="rh-url" href="{environment.url('/settings/rbac/groups/detail/' + action.events[0].payload.uuid)}"><b>{action.events[0].payload.name}</b></a>
```
Comment (gwenneg) [inline engine/src/main/resources/templates/Rbac/customPlatformGroupUpdatedEmailBody.html:48]:
```suggestion
                        Check the updated group <a class="rh-url" href="{environment.url('/settings/rbac/groups/detail/')}{action.events[0].payload.uuid}"><b>{action.events[0].payload.name}</b></a>
```

File: engine/src/main/resources/templates/Rbac/customRoleCreatedEmailBody.html [hunk 7]
```diff
@@ -35,15 +35,10 @@
                         Hi,
                         <br>
                         <br>
-                        {#if action.events[0].payload.system == "true"}Red Hat{#else}Custom{/if} role is {#if action.events[0].payload.system == "true"} {action.events[0].payload.operation}{#else} available{/if}.
+                        A new custom role has been created by {action.events[0].payload.username}.
                         <br>
-                        {#if action.events[0].payload.operation != "delete"} Check the role <a class="rh-url" href="https://console.redhat.com/settings/rbac/roles/detail/{action.events[0].payload.uuid}/"><b>{action.events[0].payload.name}</b></a>{/if}
+                        Check the created role <a class="rh-url" href="{environment.url('/settings/rbac/roles/detail/' + action.events[0].payload.uuid)}"><b>{action.events[0].payload.name}</b></a>
```
Comment (gwenneg) [inline engine/src/main/resources/templates/Rbac/customRoleCreatedEmailBody.html:40]:
```suggestion
                        Check the created role <a class="rh-url" href="{environment.url('/settings/rbac/roles/detail/')}{action.events[0].payload.uuid}"><b>{action.events[0].payload.name}</b></a>
```

File: engine/src/main/resources/templates/Rbac/customRoleUpdatedEmailBody.html [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+            <!-- Content block -->
+            <tr>
+                <td class="rh-content__block">
+                    <p>
+                        Hi,
+                        <br>
+                        <br>
+                        A custom role has been updated by {action.events[0].payload.username}.
+                        <br>
+                        Check the updated role <a class="rh-url" href="{environment.url('/settings/rbac/roles/detail/' + action.events[0].payload.uuid)}"><b>{action.events[0].payload.name}</b></a>
```
Comment (gwenneg) [inline engine/src/main/resources/templates/Rbac/customRoleUpdatedEmailBody.html:40]:
```suggestion
                        Check the updated role <a class="rh-url" href="{environment.url('/settings/rbac/roles/detail/')}{action.events[0].payload.uuid}"><b>{action.events[0].payload.name}</b></a>
```

File: engine/src/main/resources/templates/Rbac/nonPlatformRoleUpdatedEmailBody.html [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+        <table class="rh-content" role="presentation" align="center" border="0" cellpadding="0" cellspacing="0" width="100%">
+            <tbody>
+            <!-- Content block -->
+            <tr>
+                <td class="rh-content__block">
+                    <p>
+                        Hi,
+                        <br>
+                        <br>
+                        Red Hat has updated the role <a class="rh-url" href="{environment.url('/settings/rbac/roles/detail/' + action.events[0].payload.uuid)}"><b>{action.events[0].payload.name}</b></a>.
```
Comment (gwenneg) [inline engine/src/main/resources/templates/Rbac/nonPlatformRoleUpdatedEmailBody.html:38]:
```suggestion
                        Red Hat has updated the role <a class="rh-url" href="{environment.url('/settings/rbac/roles/detail/')}{action.events[0].payload.uuid}"><b>{action.events[0].payload.name}</b></a>.
```

File: engine/src/main/resources/templates/Rbac/platformGroupToCustomEmailBody.html [hunk 10]
```diff
@@ -35,9 +35,9 @@
                         Hi,
                         <br>
                         <br>
-                        Platform default group is modified and the user is responsible for managing it from now on.
+                        Platform default group is modified by {action.events[0].payload.username} and the Red Hat will not be responsible for managing it from now on.
                         <br>
-                        Check the custom platform default group here: <a class="rh-url" href="https://console.redhat.com/settings/rbac/groups/detail/{action.events[0].payload.uuid}/"><b>{action.events[0].payload.name}</b></a>
+                        Check the custom platform default group here: <a class="rh-url" href="{environment.url('/settings/rbac/groups/detail/' + action.events[0].payload.uuid)}"><b>{action.events[0].payload.name}</b></a>
```
Comment (gwenneg) [inline engine/src/main/resources/templates/Rbac/platformGroupToCustomEmailBody.html:40]:
```suggestion
                        Check the custom platform default group here: <a class="rh-url" href="{environment.url('/settings/rbac/groups/detail/')}{action.events[0].payload.uuid}"><b>{action.events[0].payload.name}</b></a>
```

File: engine/src/main/resources/templates/Rbac/platformRoleUpdatedEmailBody.html [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+        <table class="rh-content" role="presentation" align="center" border="0" cellpadding="0" cellspacing="0" width="100%">
+            <tbody>
+            <!-- Content block -->
+            <tr>
+                <td class="rh-content__block">
+                    <p>
+                        Hi,
+                        <br>
+                        <br>
+                        Red Hat has updated the role <a class="rh-url" href="{environment.url('/settings/rbac/roles/detail/' + action.events[0].payload.uuid)}"><b>{action.events[0].payload.name}</b></a>.
```
Comment (gwenneg) [inline engine/src/main/resources/templates/Rbac/platformRoleUpdatedEmailBody.html:38]:
```suggestion
                        Red Hat has updated the role <a class="rh-url" href="{environment.url('/settings/rbac/roles/detail/')}{action.events[0].payload.uuid}"><b>{action.events[0].payload.name}</b></a>.
```

File: engine/src/main/resources/templates/Rbac/roleAddedToPlatformGroupEmailBody.html [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+        <table class="rh-content" role="presentation" align="center" border="0" cellpadding="0" cellspacing="0" width="100%">
+            <tbody>
+            <!-- Content block -->
+            <tr>
+                <td class="rh-content__block">
+                    <p>
+                        Hi,
+                        <br>
+                        <br>
+                        Red Hat added a role <a class="rh-url" href="{environment.url('/settings/rbac/roles/detail/' + action.events[0].payload.role.uuid)}"><b>{action.events[0].payload.role.name}</b></a> to platform default access group.
```
Comment (gwenneg) [inline engine/src/main/resources/templates/Rbac/roleAddedToPlatformGroupEmailBody.html:38]:
```suggestion
                        Red Hat added a role <a class="rh-url" href="{environment.url('/settings/rbac/roles/detail/')}{action.events[0].payload.role.uuid}"><b>{action.events[0].payload.role.name}</b></a> to platform default access group.
```

File: engine/src/main/resources/templates/Rbac/roleRemovedFromPlatformGroupEmailBody.html [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+            <!-- Content block -->
+            <tr>
+                <td class="rh-content__block">
+                    <p>
+                        Hi,
+                        <br>
+                        <br>
+                        Red Hat removed a role from platform default access group.
+                        <br>
+                        The removed role is: <a class="rh-url" href="{environment.url('/settings/rbac/roles/detail/' + action.events[0].payload.role.uuid)}"><b>{action.events[0].payload.role.name}</b></a>
```
Comment (gwenneg) [inline engine/src/main/resources/templates/Rbac/roleRemovedFromPlatformGroupEmailBody.html:40]:
```suggestion
                        The removed role is: <a class="rh-url" href="{environment.url('/settings/rbac/roles/detail/')}{action.events[0].payload.role.uuid}"><b>{action.events[0].payload.role.name}</b></a>
```

File: engine/src/main/resources/templates/Rbac/systemRoleAvailableEmailBody.html [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+            <!-- Content block -->
+            <tr>
+                <td class="rh-content__block">
+                    <p>
+                        Hi,
+                        <br>
+                        <br>
+                        Red Hat now provides a new role.
+                        <br>
+                        Check the new available role <a class="rh-url" href="{environment.url('/settings/rbac/roles/detail/' + action.events[0].payload.uuid)}"><b>{action.events[0].payload.name}</b></a>
```
Comment (gwenneg) [inline engine/src/main/resources/templates/Rbac/systemRoleAvailableEmailBody.html:40]:
```suggestion
                        Check the new available role <a class="rh-url" href="{environment.url('/settings/rbac/roles/detail/')}{action.events[0].payload.uuid}"><b>{action.events[0].payload.name}</b></a>
```

File: engine/src/main/resources/templates/Rbac/customRoleUpdatedEmailTitle.txt [hunk 15]
```diff
@@ -0,0 +1 @@
+{action.timestamp.toUtcFormat()} - Custom role {action.events[0].payload.name} is updated.
```
Comment (gwenneg) [inline engine/src/main/resources/templates/Rbac/customRoleUpdatedEmailTitle.txt:1]: `is updated` or `has been updated`?

File: engine/src/main/resources/templates/Rbac/customGroupCreatedEmailBody.html (hunk 2 above)
Comment (astrozzc) [reply engine/src/main/resources/templates/Rbac/customGroupCreatedEmailBody.html]: Good to know, thanks!

**Comment by gwenneg** [comment]
The test failure isn't directly related to your PR. We're in the middle of a big change in the templates area and the failing test validates some migration code. I'll fix it for you in this PR.

**Comment by astrozzc** [comment]
> The test failure isn't directly related to your PR. We're in the middle of a big change in the templates area and the failing test validates some migration code. I'll fix it for you in this PR.

Got it, thank you!

**Review by gwenneg — APPROVED** [review]
LGTM

**Pull request merged**